cmdLineTester_jfr_0 WorkLoad.generateClassLoader ArrayIndexOutOfBoundsException

https://openj9-jenkins.osuosl.org/job/Test_openjdk11_j9_sanity.functional_aarch64_linux_Nightly_testList_0/1014
cmdLineTester_jfr_0
```
00:04:15  Testing: runWorkload - approx 30seconds
00:04:15  Test start time: 2025/10/25 04:04:15 Coordinated Universal Time
00:04:15  Running command: "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_aarch64_linux_Nightly_testList_0/jdkbinary/j2sdk-image/bin/java"   -XX:StartFlightRecording --add-exports java.base/com.ibm.oti.vm=ALL-UNNAMED -cp /home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_aarch64_linux_Nightly_testList_0/aqa-tests/TKG/../../jvmtest/functional/cmdLineTests/jfr/jfr.jar org.openj9.test.WorkLoad

...
'00:55:26   [ERR] Exception in thread "Thread-56" java.lang.RuntimeException: java.lang.ArrayIndexOutOfBoundsException: Array index out of range: 16
00:55:26   [ERR] 	at org.openj9.test.WorkLoad.generateClassLoader(WorkLoad.java:204)
00:55:26   [ERR] 	at org.openj9.test.WorkLoad.workload(WorkLoad.java:108)
00:55:26   [ERR] 	at org.openj9.test.WorkLoad.lambda$runWork$0(WorkLoad.java:81)
00:55:26   [ERR] 	at java.base/java.lang.Thread.run(Thread.java:836)
00:55:26   [ERR] Caused by: java.lang.ArrayIndexOutOfBoundsException: Array index out of range: 16
00:55:26   [ERR] 	at java.base/java.util.ArrayList.add(ArrayList.java:487)
00:55:26   [ERR] 	at java.base/java.util.ArrayList.add(ArrayList.java:499)
00:55:26   [ERR] 	at org.openj9.test.WorkLoad.generateClassLoader(WorkLoad.java:201)
00:55:26   [ERR] 	... 3 more
```

https://openj9-jenkins.osuosl.org/job/Test_openjdk11_j9_sanity.functional_x86-64_linux_Nightly_testList_0/943
cmdLineTester_jfr_0
```
00:12:16  Testing: runWorkload 2 - approx 2mins
00:12:16  Test start time: 2025/10/25 00:12:14 Eastern Standard Time
00:12:16  Running command: "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Nightly_testList_0/jdkbinary/j2sdk-image/bin/java"   -XX:StartFlightRecording --add-exports java.base/com.ibm.oti.vm=ALL-UNNAMED -cp /home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_Nightly_testList_0/aqa-tests/TKG/../../jvmtest/functional/cmdLineTests/jfr/jfr.jar org.openj9.test.WorkLoad 200 20000 200

...
01:03:40   [ERR] Exception in thread "Thread-67" java.lang.RuntimeException: java.lang.ArrayIndexOutOfBoundsException
01:03:40   [ERR] 	at org.openj9.test.WorkLoad.generateClassLoader(WorkLoad.java:204)
01:03:40   [ERR] 	at org.openj9.test.WorkLoad.workload(WorkLoad.java:108)
01:03:40   [ERR] 	at org.openj9.test.WorkLoad.lambda$runWork$0(WorkLoad.java:81)
01:03:40   [ERR] 	at java.base/java.lang.Thread.run(Thread.java:836)
01:03:40   [ERR] Caused by: java.lang.ArrayIndexOutOfBoundsException
01:03:40   [ERR] 	at java.base/java.util.ArrayList.add(ArrayList.java:487)
01:03:40   [ERR] 	at java.base/java.util.ArrayList.add(ArrayList.java:499)
01:03:40   [ERR] 	at org.openj9.test.WorkLoad.generateClassLoader(WorkLoad.java:201)
01:03:40   [ERR] 	... 3 more
```

Changes from previous build:
https://github.com/eclipse-openj9/openj9/compare/9726848668...873e4524ef9

--- a/test/functional/cmdLineTests/jfr/src/org/openj9/test/WorkLoad.java
+++ b/test/functional/cmdLineTests/jfr/src/org/openj9/test/WorkLoad.java
@@ -27,24 +27,29 @@ import java.net.URLClassLoader;
 import java.nio.file.Paths;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Queue;
 import java.util.concurrent.Callable;
+import java.util.concurrent.ConcurrentLinkedQueue;
 import java.util.concurrent.atomic.AtomicLong;
 import java.util.concurrent.locks.LockSupport;
 
 public class WorkLoad {
-	private int numberOfThreads;
-	private int sizeOfNumberList;
-	private int repeats;
-	private boolean vthreads;
+
+	private static interface GlobalLoack { }
 
 	public static double average;
 	public static double stdDev;
 
-	static ArrayList<Class<?>> classes = new ArrayList<>();
+	private static final Queue<Class<?>> classes = new ConcurrentLinkedQueue<>();
 
 	private static long globalCounter = 0;
-	static interface GlobalLoack {}
-	private static Object globalLock = new GlobalLoack(){};
+
+	private static Object globalLock = new GlobalLoack() { };
+
+	private final int numberOfThreads;
+	private final int sizeOfNumberList;
+	private final int repeats;
+	private final boolean vthreads;
 
 	public WorkLoad(int numberOfThreads, int sizeOfNumberList, int repeats, boolean vthreads) {
 		this.numberOfThreads = numberOfThreads;
@@ -111,23 +116,19 @@ public class WorkLoad {
 
 			Class<?> threadClass = Thread.class;
 			Method ofVirtualMethod = threadClass.getMethod("ofVirtual");
-
 			Object virtualThreadBuilder = ofVirtualMethod.invoke(null);
 
-			Method unstartedMethod = virtualThreadBuilder.getClass()
-				.getMethod("unstarted", Runnable.class);
+			Class<?> builderClass = Class.forName("java.lang.Thread$Builder");
+			Method unstartedMethod = builderClass.getMethod("unstarted", Runnable.class);
 
+			Runnable task = () -> {
+				burnCPU();
+				generateTimedPark();
+				burnCPU();
+				completionCount.incrementAndGet();
+			};
 
-			ofVirtualMethod.setAccessible(true);
-			unstartedMethod.setAccessible(true);
 			for (int i = 0; i < numVirtualThreads; i++) {
-				Runnable task = () -> {
-					burnCPU();
-					generateTimedPark();
-					burnCPU();
-					completionCount.incrementAndGet();
-				};
-
 				Thread vthread = (Thread) unstartedMethod.invoke(virtualThreadBuilder, task);
 				threads[i] = vthread;
 				vthread.start();
@@ -136,7 +137,6 @@ public class WorkLoad {
 			e.printStackTrace();
 		}
 	}
-
 
 	private void workload() {
 		for (int i = 0; i < repeats; i++) {
@@ -178,7 +178,9 @@ public class WorkLoad {
 		if (0 == depth) {
 			try {
 				c.call();
-			} catch (Exception e) {}
+			} catch (Exception e) {
+				// ignore
+			}
 			return;
 		}
 
@@ -193,7 +195,9 @@ public class WorkLoad {
 
 		try {
 			c.call();
-		} catch (Exception e) {}
+		} catch (Exception e) {
+			// ignore
+		}
 	}
 
 	private void generateTimedPark() {
@@ -214,7 +218,7 @@ public class WorkLoad {
 		recursiveFucntionWithCallable(10, () -> {
 			Object o = new Object();
 
-			synchronized(o) {
+			synchronized (o) {
 				o.wait(100);
 			}
 
@@ -223,7 +227,7 @@ public class WorkLoad {
 	}
 
 	private void burnCPU() {
-		ArrayList<Double> numberList = new ArrayList<Double>();
+		List<Double> numberList = new ArrayList<>();
 
 		for (int i = 0; i < sizeOfNumberList; i++) {
 			numberList.add(Math.random());
@@ -234,13 +238,12 @@ public class WorkLoad {
 		stdDev += standardDeviation(numberList);
 	}
 
-	public static class ClassLoaderTestClass {}
+	static final class ClassLoaderTestClass { }
 
 	private void emitDataLoss() {
 		try {
 			Class<?> jvmClass = Class.forName("jdk.jfr.internal.JVM");
-			Method emitDataLossMethod = jvmClass.getDeclaredMethod("emitDataLoss", long.class);
-			emitDataLossMethod.setAccessible(true);
+			Method emitDataLossMethod = jvmClass.getMethod("emitDataLoss", long.class);
 			emitDataLossMethod.invoke(null, 1024L);
 		} catch (Exception e) {
 			e.printStackTrace();
@@ -286,11 +289,15 @@ public class WorkLoad {
 	private void throwThrowable(Class<? extends Throwable> throwable) {
 		try {
 			throw throwable.getDeclaredConstructor().newInstance();
-		} catch (Throwable t) {}
+		} catch (Throwable t) {
+			// ignore
+		}
 
 		try {
 			throw throwable.getDeclaredConstructor().newInstance("random string: " + Math.random());
-		} catch (Throwable t) {}
+		} catch (Throwable t) {
+			// ignore
+		}
 	}
 
 	private void throwThrowables() {
@@ -301,4 +308,5 @@ public class WorkLoad {
 		throwThrowable(ClassCastException.class);
 		throwThrowable(NumberFormatException.class);
 	}
+
 }

